### DISCUSSION TITLE
Show full site tree structure / add ability to show full site tree structure

### DISCUSSION THREAD

**Discussion by ghost** [requarks/wiki#5327] [General]
Hello there,

We're currently migrating from dokuwiki to wikijs.

One thing that's really bothering me is the navigation. Right now when using site tree navigation you have to click through folders to see their contents. I'd like the ability to see the whole structure.

A good example would be the pages structure in Atlassian Confluence. You create pages and subpages, and you can see the structure.

![image](https://user-images.githubusercontent.com/5992169/170211303-bd851392-c28e-4f03-8495-5eb9cf5e716e.png)
(snippet from Atlassian website)

Right now everything is hidden, which is unintuitive.

Thank you.

**Comment by vladyxa13** [comment]
Will be cool also, if user can insert something like tree into root page.
Like if we have catalog named 'vegetables' and child elements: 'potato', 'carrot'. If we put 'tree tool' to page 'vegetables' it will be show pages under this 'vegetables', it will be show 'potato', 'carrot'.


Did u find anyone who can help with your variant?

**Comment by imanmagomedov** [comment]
really need that

**Comment by CemraJC** [comment]
I ended up here after searching for page tree functionality, and although this isn't necessarily what you were after, someone else may find this and find it useful. 

If you're happy with a quickly thrown together script, chuck this gist into a wiki page (put the JS file in the page's "Scripts" tab, or it won't execute) and you'll get a full tree generated of all the pages on the site. I was too impatient to wait for version 3 to come out: https://gist.github.com/CemraJC/8438a616740162661a9958c46c95315d

**Reply by CemraJC** [reply]
Note, I created a separate discussion for this, seems more appropriate: https://github.com/requarks/wiki/discussions/6982

**Comment by johnhaire89** [comment]
I noticed that there's this PR from 2022 that hasn't been merged https://github.com/requarks/wiki/pull/5340
Dev has even hosted a demo at https://wiki.deepin.org/zh/home
